Refactor tests

ravisantoshgudimetla opened the pull request:
Small refactor to check if framework can be  re-used.

ravisantoshgudimetla commented:
If this passes, l'll remove the wsu/e2e_test.go file.

aravindhp commented on internal/test/framework/framework.go:
```diff
@@ -1,26 +1,16 @@
 package test
 
 import (
-	"flag"
```
Why does dep updates have code changes? I will review the rest once this is resolved.

aravindhp commented on internal/test/go.mod:
```diff
@@ -2,13 +2,19 @@ module github.com/openshift/windows-machine-config-operator/internal/test
 
 go 1.12
 
+replace (
```
If you ran a command to do this, please mention that in the commit.

ravisantoshgudimetla replied:
No, I just manually added them

aravindhp commented:
/hold

ravisantoshgudimetla commented:
/retest

aravindhp commented on hack/run-wmcb-ci-e2e-test.sh:
```diff
@@ -5,14 +5,13 @@ set -o pipefail
-CGO_ENABLED=0 GO111MODULE=on go test -c wmcb_test.go -o wmcb_framework
-./wmcb_framework -test.run=TestWMCBUnit -binaryToBeTransferred=../../wmcb_unit_test.exe
+cd "${WMCB_TEST_DIR}"
+CGO_ENABLED=0 GO111MODULE=on go test -v -run=TestWMCBUnit -binaryToBeTransferred=../../../wmcb_unit_test.exe -timeout=20m .
```
@sebsoto bumped the timeout to 30m. I suggest we do the same here too.

aravindhp commented on internal/test/framework/framework.go:
```diff
@@ -0,0 +1,198 @@
+	cloudProvider cloudprovider.Cloud
+	// k8sclientset is the kubernetes clientset we will use to query the cluster's status
+	K8sclientset *kubernetes.Clientset
+}
```
Do all the fields in the struct need to exported i.e. capitalized?

ravisantoshgudimetla replied:
Not all of them, for example `cloudProvider` doesn't need to be exported. But as I told at https://github.com/openshift/windows-machine-config-operator/pull/103/files#diff-5465ccc52b3a24461a98f029910cf60aR29, this is not clear, ideally we should have getters and setters around this

aravindhp commented on internal/test/framework/framework.go:
```diff
@@ -0,0 +1,198 @@
+	// Use Windows 2019 server image with containers in us-east1 zone for CI testing.
+	// TODO: Move to environment variable that can be fetched from the cloud provider
+	imageID := "ami-0b8d82dea356226d3"
+	instanceType := "m4.large"
```
Please see #106 and use `m5a.large`

ravisantoshgudimetla replied:
Thanks for catching it, I thought I already did this while doing rebase but obviously not and this could be causing the failures we're noticing in CI

aravindhp commented on internal/test/framework/framework.go:
```diff
@@ -0,0 +1,198 @@
+		os.Stdout, os.Stderr); err != nil {
+		return fmt.Errorf("failed to start sshd with error %v", err)
+	}
+	return nil
```
In the above message, please replace `with error` with `:` to make the message more succinct.

ravisantoshgudimetla replied:
Done

ravisantoshgudimetla commented:
> Sorry for not catching this before. Please replace [test] with [ci] in your commit message.

No problem, I just updated it. I started with `[ci]` but it did not make much sense as I thought this is not specific just specific to CI so I changed it to `test`. I've changed it now.

ravisantoshgudimetla commented:
/retest

ravisantoshgudimetla commented:
/retest

ravisantoshgudimetla commented:
/approve cancel

ravisantoshgudimetla commented:
/retest

sebsoto commented on internal/test/wsu/wsu_test.go:
```diff
@@ -178,25 +101,25 @@ func TestWSU(t *testing.T) {
-		_, err := winRMClient.Run(command, stdout, os.Stderr)
+
+		_, err := framework.WinrmClient.Run(command, stdout, os.Stderr)
+
```
Is this newline intentional?

ravisantoshgudimetla replied:
No, when I was doing rebase, I accidentally messed up the newlines. I am reverting those changes. Thanks for catching it

sebsoto commented on internal/test/wsu/wsu_test.go:
```diff
@@ -103,69 +70,25 @@ ansible_winrm_server_cert_validation=ignore`, ip, password, clusterAddress))
 	out, err := cmd.CombinedOutput()
 	require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, string(out))
 
+	k8sclientset = framework.K8sclientset
```
Shouldnt we delete the k8sclientset and just use framework.K8sclientset at this point?

sebsoto commented on internal/test/framework/framework.go:
```diff
@@ -0,0 +1,198 @@
+	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
+	// has those services present
+	time.Sleep(time.Minute)
+	if err := f.configureOpenSSHServer(); err != nil {
```
This is only needed for the WMCB tests, should it be a shared step?

sebsoto commented on internal/test/framework/framework.go:
```diff
@@ -0,0 +1,198 @@
+	if err := f.configureOpenSSHServer(); err != nil {
+		log.Fatalf("failed to configure OpenSSHServer on the Windows VM: %v", err)
+	}
+	if err := f.createRemoteDir(); err != nil {
```
Same as above

sebsoto commented on internal/test/framework/framework.go:
```diff
@@ -0,0 +1,198 @@
+	if err := f.createRemoteDir(); err != nil {
+		log.Fatalf("failed to create remote dir with error: %v", err)
+	}
+	if err := f.getSSHClient(); err != nil {
```
Same as above

sebsoto commented on internal/test/framework/framework.go:
```diff
@@ -0,0 +1,198 @@
+	K8sclientset *kubernetes.Clientset
+}
+
+// Setup sets up the initial test cluster for the Windows
```
This comment is confusing as we arent setting up a cluster here

ravisantoshgudimetla commented:
/retest

sebsoto commented:
/lgtm
/hold cancel
Nice work @ravisantoshgudimetla !
